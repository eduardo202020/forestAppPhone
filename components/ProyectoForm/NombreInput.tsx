import React from 'react';
import { TextInput, Text, StyleSheet } from 'react-native';

interface NombreInputProps {
  value: string;
  onChange: (v: string) => void;
  editable?: boolean;
  error?: string;
}

const NombreInput: React.FC<NombreInputProps> = ({
  value,
  onChange,
  editable = true,
  error,
}) => (
  <>
    <TextInput
      style={styles.input}
      placeholder="Nombre del proyecto"
      value={value}
      onChangeText={onChange}
      editable={editable}
    />
    {error && <Text style={styles.errorText}>{error}</Text>}
  </>
);

const styles = StyleSheet.create({
  input: {
    borderWidth: 1,
    borderColor: '#ccc',
    borderRadius: 8,
    padding: 10,
    marginBottom: 12,
    fontSize: 16,
    backgroundColor: '#f9f9f9',
  },
  errorText: {
    color: 'red',
    marginBottom: 8,
    fontSize: 13,
  },
});

export default NombreInput;
