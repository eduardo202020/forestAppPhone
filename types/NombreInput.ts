export interface NombreInputProps {
  value: string;
  onChange: (value: string) => void;
  editable?: boolean;
  error?: string;
}
