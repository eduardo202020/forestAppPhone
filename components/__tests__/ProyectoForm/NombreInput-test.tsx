import React from 'react';
import renderer from 'react-test-renderer';
import NombreInput from '../../ProyectoForm/NombreInput';

describe('NombreInput', () => {
  it('se renderiza sin errores', () => {
    const tree = renderer
      .create(
        <NombreInput
          value="Proyecto A"
          onChange={() => {}}
          editable
          error={undefined}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('muestra el error si existe', () => {
    const tree = renderer
      .create(
        <NombreInput
          value=""
          onChange={() => {}}
          editable
          error="Campo requerido"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
