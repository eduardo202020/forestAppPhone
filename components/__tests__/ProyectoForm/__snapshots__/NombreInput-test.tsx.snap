// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NombreInput muestra el error si existe 1`] = `
[
  <TextInput
    editable={true}
    onChangeText={[Function]}
    placeholder="Nombre del proyecto"
    style={
      {
        "borderColor": "#ccc",
        "borderRadius": 5,
        "borderWidth": 1,
        "marginBottom": 10,
        "padding": 10,
      }
    }
    value=""
  />,
  <Text
    style={
      {
        "color": "red",
        "fontSize": 12,
      }
    }
  >
    Campo requerido
  </Text>,
]
`;

exports[`NombreInput se renderiza sin errores 1`] = `
<TextInput
  editable={true}
  onChangeText={[Function]}
  placeholder="Nombre del proyecto"
  style={
    {
      "borderColor": "#ccc",
      "borderRadius": 5,
      "borderWidth": 1,
      "marginBottom": 10,
      "padding": 10,
    }
  }
  value="Proyecto A"
/>
`;
